src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs

```diff
@@ -104,7 +107,7 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             using (JsonTextWriter writer = new JsonTextWriter(sw) { Formatting = Formatting.None })
             {
                 writer.WriteStartObject();
-                WritePropertyIfNotNull(writer, "appName", _appServiceOptions.CurrentValue.AppName);
+                WritePropertyIfNotNull(writer, "appName", _appServiceOptions.AppName);
```

Comment:
Given that this introduces the risk of stale reads as well as updates to the options happening concurrent to the reads can that be problematic?

Reply:
In practice, there's little to no race difference here. For example, if you called `.CurrentValue.X` 5 times in a row (see `SystemLogger` for this actually happening), you'll already tear work across the reference boundary and get skew during a single function call. You can work around this with `IOptionsSnapshot` (which allocates a copy per scope) or getting a ref and re-using once in the method, but the reality is somewhere in a request pipe between methods, things will change and that should be expected if options are _ever_ changing.

I'm not sure the phrasing is awesome there, but the baseline is: yep, it's expected values may change as you go. That's why you should try to avoid:

```cs
if (setting.Thing != null)
{
    var a = setting.Thing.Something();
}
```
...because after that `if` check, it may _become_ null if volatile/reloadable. Instead, strive for:
```cs
var a = setting.Thing?.Something();
```
...or:
```cs
var snapshot = setting.Thing;
if (snapshot != null)
{
    var a = snapshot.Something();
}
```
In general, be aware that these can change _literally at any time_ and plan accordingly, especially if nullable.